entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md

```diff
@@ -14,6 +14,137 @@ This page does not duplicate the [plan for EF Core 5.0](xref:core/what-is-new/ef
+
+This creates problems for types where the CLR default is not a good sentinel--most notably, `bool` properties. EF Core 5.0 now allows the backing field to be nullable for cases like this. For example:
+
+```csharp
+public class Blog
+{
+    private bool? _isValid;
+
+    public bool IsValid
+    {
+        get => _isValid ?? false;
+        set => _isValid = value;
+    }
+}
+```
+
+Note that the backing field is nullable, but the publicly exposed property is not. This allows the sentinel value to be `null` without impacting the public surface of the entity type. In this case, if the `IsValid` is never set, then the database default will be used since the backing field remains null. If either `true` or `false` are set, then this value is saved explicitly to the database.
+
+Documentation is tracked by issue [#2525](https://github.com/dotnet/EntityFramework.Docs/issues/2525).
+
+### Cosmos partition keys
+
+EF Core allows the Cosmos partition key is included in the EF model. For example:
+
+```csharp
+modelBuilder.Entity<Customer>().HasPartitionKey(b => b.AlternateKey)
+```
+
+Starting with preview 7, the partition key is included in the entity type's PK and is used to improved performance in some queries.
+
+Documentation is tracked by issue [#2471](https://github.com/dotnet/EntityFramework.Docs/issues/2471).
+
+### Cosmos configuration
+
+EF Core 5.0 improves configuration of Cosmos and Cosmos connections.
+
+Previously, EF Core required the end-point and key to be specified explicitly when connecting to a Cosmos database. EF Core 5.0 allows use of a connection string instead. In addition, EF Core 5.0 allows the WebProxy instance to be explicitly set. For example:
+
+```csharp
+protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    => optionsBuilder
+        .UseCosmos("my-cosmos-connection-string", "MyDb",
+            cosmosOptionsBuilder =>
+            {
+                cosmosOptionsBuilder.WebProxy(myProxyInstance);
+            });
+```
+
+Many other timeout values, limits, etc. can now also be configured. For example:
+
+```csharp
+protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    => optionsBuilder
+        .UseCosmos("my-cosmos-connection-string", "MyDb",
+            cosmosOptionsBuilder =>
+            {
+                cosmosOptionsBuilder.LimitToEndpoint();
+                cosmosOptionsBuilder.AllowBulkExecution();
+                cosmosOptionsBuilder.RequestTimeout(requestTimeout);
+                cosmosOptionsBuilder.OpenTcpConnectionTimeout(timeout);
+                cosmosOptionsBuilder.IdleTcpConnectionTimeout(timeout);
+                cosmosOptionsBuilder.PortReuseMode(portReuseMode);
+                cosmosOptionsBuilder.EnableTcpConnectionEndpointRediscovery();
```

Comment:
AllowBulkExecution, PortReuseMode and EnableTcpConnectionEndpointRediscovery were removed in the 4.0 upgrade